File: easy/e121/python3/solution_test.py
import solution


def test() -> str:
    s = solution.Solution()

    input1 = [[6, 1, 3, 2, 4, 7], [1, 4, 2], [3, 3, 5, 0, 0, 3, 1, 4], [7, 1, 5, 3, 6, 4], [
        7, 6, 4, 3, 1], [2, 7, 1, 4], [3, 2, 6, 5, 0, 3]]

    expectations = [6, 3, 4, 5, 0, 5, 4]

    if (len(input1) != len(expectations)):
        return "test is invalid"

    for i in range(len(expectations)):
        assert (1 <= len(input1[i]) and len(input1[i]) < 10**5)

        for j in range(len(input1[i])):
            assert (0 <= input1[i][j] and input1[i][j] <= 10**4)

        origInput1 = input1[i][:]

        result = s.maxProfit(input1[i])

        failMsg = (f"bad result for {origInput1}: got {
                   result} expected {expectations[i]}")

        if result != expectations[i]:
            return failMsg

    return "PASS"


if __name__ == '__main__':
    print(test())

File: easy/e2974/cpp/solution_test.cpp
#include "solution.h"

#include <cassert>
#include <iostream>
#include <vector>

int main() {
    std::vector<std::vector<int>> input1 = { {5,4,2,3}, {2,5}, {5,4,2,3,6,1,7,8}, {4,4,3,8} };

    std::vector<std::vector<int>> expectations = { {3,2,5,4}, {5,2}, {2,1,4,3,6,5,8,7}, {4,3,8,4} };

    if (input1.size() != expectations.size()) {
        std::cerr << "test is invalid";
        return 0;
    }

    Solution s;
    for (int i = 0; i < input1.size(); i++) {
        assert(1 <= input1[i].size() && input1[i].size() <= 100);
        assert(input1[i].size() % 2 == 0);

        std::vector<int> origInput1(input1[i].size());
        for (int j = 0; j < origInput1.size(); j++) {
            assert(1 <= input1[i][j] && input1[i][j] <= 100);
            origInput1[j] = input1[i][j];
        }

        std::vector<int> result = s.numberGame(input1[i]);

        if (result != expectations[i]) {
            std::cerr << "FAIL: bad result for [";

            for (int j = 0; j < origInput1.size(); j++) {
                if (j > 0) {
                    std::cerr << ", ";
                }

                std::cerr << origInput1[j];
            }

            std::cerr << "]: got [";

            for (int j = 0; j < result.size(); j++) {
                if (j > 0) {
                    std::cerr << ", ";
                }

                std::cerr << result[j];
            }

            std::cerr << "] expected [";

            for (int j = 0; j < expectations[i].size(); j++) {
                if (j > 0) {
                    std::cerr << ", ";
                }

                std::cerr << expectations[i][j];
            }

            std::cerr << "]" << std::endl;

            return 0;
        }
    }

    std::cout << "PASS";

    return 0;
}